homeassistant/components/syncthing/sensor.py

```diff
@@ -0,0 +1,224 @@
+        self._folder_id = folder_id
+        self._folder_label = folder_label
+        self._state = None
+        self._unsub_timer = None
+        self._version = version
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return f"{self._name} {self._folder_id} {self._folder_label}"
+
+    @property
+    def unique_id(self):
+        """Return the unique id of the entity."""
+        return f"{DOMAIN}-{self._name}-{self._folder_id}"
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._state["state"]
+
+    @property
+    def available(self):
+        """Could the device be accessed during the last update call."""
+        return self._state is not None
+
+    @property
+    def icon(self):
+        """Return the icon for this sensor."""
+        if self._state is None:
+            return FOLDER_SENSOR_DEFAULT_ICON
+        if self.state in FOLDER_SENSOR_ICONS:
+            return FOLDER_SENSOR_ICONS[self.state]
+        return FOLDER_SENSOR_ALERT_ICON
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes."""
+        return self._state
+
+    @property
+    def should_poll(self):
+        """Return the polling requirement for this sensor."""
+        return False
+
+    @property
+    def device_info(self):
+        """Return device information."""
+        return {
+            "identifiers": {(DOMAIN, self._name)},
+            "name": self._name,
+            "manufacturer": "Syncthing Team",
+            "sw_version": self._version,
+            "entry_type": "service",
+        }
+
+    async def async_update_status(self):
+        """Request folder status and update state."""
+        try:
+            state = await self._syncthing.database.status(self._folder_id)
+            # A workaround, for some reason, state of paused folders is an empty string
+            if state["state"] == "":
+                state["state"] = "paused"
+            self._state = state
+            self._state.update({"id": self._folder_id})
+        except aiosyncthing.exceptions.SyncthingError:
+            self._state = None
+        self.async_write_ha_state()
+
+    def subscribe(self):
+        """Start polling syncthing folder status."""
+        if self._unsub_timer is None:
+
+            async def refresh(event_time):
+                """Get the latest data from Syncthing."""
+                await self.async_update_status()
+
+            self._unsub_timer = async_track_time_interval(
+                self.hass, refresh, SCAN_INTERVAL
+            )
+
+    def unsubscribe(self):
+        """Stop polling syncthing folder status."""
+        if self._unsub_timer is not None:
+            self._unsub_timer()
+            self._unsub_timer = None
+
+    async def async_added_to_hass(self):
+        """Handle entity which will be added."""
+
+        async def handle_folder_summary(event):
```

Comment:
This can be a callback since we don't await inside.
```suggestion
        @callback
        def handle_folder_summary(event):
```